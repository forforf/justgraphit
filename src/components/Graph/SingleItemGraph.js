import React from 'react';
import relative from '../../datetime/relative'

const style = {
  marginLeft: '5em',
  marginTop: '2em',
  paddingTop: '2em',
  paddingBottom: '2em',
  borderStyle: 'solid'
};

const SingleItemGraph = ({itemNumber, itemDatetime}) =>
    <div style={style}>
      <div>Not enough to graph yet. Only a single lonely value of:</div>
      <h3>{itemNumber}</h3>
      {relative(itemDatetime)}
    </div>;

export default SingleItemGraph;
