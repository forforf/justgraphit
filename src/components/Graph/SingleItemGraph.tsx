import React from 'react';
import relative from '../../datetime/relative';
import { ISODatetimeType } from '../../JustGraphitTypes';

const style = {
  marginLeft: '5em',
  marginTop: '2em',
  paddingTop: '2em',
  paddingBottom: '2em',
  borderStyle: 'solid',
};

type SingleItemGraphProps = {
  itemNumber: number;
  itemDatetime: ISODatetimeType;
};

const SingleItemGraph = ({
  itemNumber,
  itemDatetime,
}: SingleItemGraphProps): JSX.Element => {
  const itemDate = new Date(itemDatetime);
  return (
    <div style={style}>
      <div>Not enough to graph yet. Only a single lonely value of:</div>
      <h3>{itemNumber}</h3>
      {relative(itemDate)}
    </div>
  );
};

export default SingleItemGraph;
