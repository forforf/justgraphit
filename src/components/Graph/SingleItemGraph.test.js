import renderer from 'react-test-renderer';
import SingleItemGraph from './SingleItemGraph';


test('deep snapshot', () => {
  const component = renderer.create(
    <SingleItemGraph itemNumber={42} itemDatetime={'2000-01-01T00:00:00.000Z'} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});