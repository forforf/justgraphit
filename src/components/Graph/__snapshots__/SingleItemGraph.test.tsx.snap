// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot 1`] = `
<div
  style={
    Object {
      "borderStyle": "solid",
      "marginLeft": "5em",
      "marginTop": "2em",
      "paddingBottom": "2em",
      "paddingTop": "2em",
    }
  }
>
  <div>
    Not enough to graph yet. Only a single lonely value of:
  </div>
  <h3>
    42
  </h3>
  23 years ago
</div>
`;
